File: DynamicKey/AgoraDynamicKey/nodejs/src/ChatTokenBuilder.js
const AccessToken = require('../src/AccessToken2').AccessToken2
const ServiceChat = require('../src/AccessToken2').ServiceChat

class ChatTokenBuilder {
    /**
     * Build the Chat user token.
     *
     * @param appId The App ID issued to you by Agora. Apply for a new App ID from
     * Agora Dashboard if it is missing from your kit. See Get an App ID.
     * @param appCertificate Certificate of the application that you registered in
     * the Agora Dashboard. See Get an App Certificate.
     * @param userUuid The user's id, must be unique.
     * @param expire represented by the number of seconds elapsed since now. If, for example, you want to access the
     * Agora Service within 10 minutes after the token is generated, set expire as 600(seconds).
     * @return The Chat User token.
     */
    static buildUserToken(appId, appCertificate, userUuid, expire) {
        const token = new AccessToken(appId, appCertificate, null, expire)
        const serviceChat = new ServiceChat(userUuid)
        serviceChat.add_privilege(ServiceChat.kPrivilegeUser, expire)
        token.add_service(serviceChat)
        return token.build()
    }

    /**
     * Build the Chat App token.
     *
     * @param appId The App ID issued to you by Agora. Apply for a new App ID from
     * Agora Dashboard if it is missing from your kit. See Get an App ID.
     * @param appCertificate Certificate of the application that you registered in
     * the Agora Dashboard. See Get an App Certificate.
     * @param expire represented by the number of seconds elapsed since now. If, for example, you want to access the
     * Agora Service within 10 minutes after the token is generated, set expire as 600(seconds).
     * @return The Chat App token.
     */
    static buildAppToken(appId, appCertificate, expire) {
        const token = new AccessToken(appId, appCertificate, null, expire)
        const serviceChat = new ServiceChat()
        serviceChat.add_privilege(ServiceChat.kPrivilegeApp, expire)
        token.add_service(serviceChat)
        return token.build()
    }
}

exports.ChatTokenBuilder = ChatTokenBuilder

File: DynamicKey/AgoraDynamicKey/dart/lib/src/chat_token_builder.dart
import 'access_token.dart';

/// Builds Token007 tokens for Chat users and applications.
class ChatTokenBuilder {
  /// Builds a Chat user token.
  ///
  /// [appId] is the App ID issued by Agora; create one in Agora Dashboard if needed.
  /// [appCertificate] is the certificate of the application registered in Agora Dashboard.
  /// [userUuid] is the user's ID and must be unique.
  /// [expireSeconds] is the expiration time in seconds from now; use `600` for 10 minutes.
  ///
  /// Returns the generated Chat user token.
  static String buildChatUserToken({
    required String appId,
    required String appCertificate,
    required String userUuid,
    required int expireSeconds,
  }) {
    final token = AccessToken.create(
      appId,
      appCertificate,
      expire: expireSeconds,
    );
    token.addService(
      ServiceChat(userUuid)
        ..addPrivilege(ServiceChat.privilegeUser, expireSeconds),
    );
    return token.build();
  }

  /// Builds a Chat application token.
  ///
  /// [appId] is the App ID issued by Agora; create one in Agora Dashboard if needed.
  /// [appCertificate] is the certificate of the application registered in Agora Dashboard.
  /// [expireSeconds] is the expiration time in seconds from now; use `600` for 10 minutes.
  ///
  /// Returns the generated Chat application token.
  static String buildChatAppToken({
    required String appId,
    required String appCertificate,
    required int expireSeconds,
  }) {
    final token = AccessToken.create(
      appId,
      appCertificate,
      expire: expireSeconds,
    );
    token.addService(
      ServiceChat()..addPrivilege(ServiceChat.privilegeApp, expireSeconds),
    );
    return token.build();
  }
}
